examples/crowdfunding/test/crowdfundingTest.js

```diff
@@ -0,0 +1,272 @@
+        appId: applicationId,
+        payFlags: {},
+        appArgs: appArgs
+      },
+      {
+        type: TransactionType.TransferAlgo,
+        sign: SignType.SecretKey,
+        fromAccount: donor.account,
+        toAccountAddr: escrow.address,
+        amountMicroAlgos: donationAmount,
+        payFlags: {}
+      }
+    ];
+    // execute transaction
+    await runtime.executeTx(txGroup, program, []);
+
+    syncAccounts();
+    assert.equal(escrow.balance(), donationAmount);
+    assert.equal(donor.balance(), initialDonorBalance - donationAmount);
+
+    // donor should be able to reclaim if goal is met
+    appArgs = [stringToBytes('reclaim')];
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    txGroup = [
+      {
+        type: TransactionType.CallNoOpSSC,
+        sign: SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: {},
+        appArgs: appArgs,
+        accounts: [escrow.address] //  AppAccounts
+      },
+      {
+        type: TransactionType.TransferAlgo,
+        sign: SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: donor.address,
+        amountMicroAlgos: 300000,
+        lsig: escrow,
+        payFlags: {}
+      }
+    ];
+
+    syncAccounts();
+    const donorBalance = donor.balance();
+    await runtime.executeTx(txGroup, program, []);
+
+    syncAccounts();
+    assert.equal(escrow.balance(), 300000);
+    assert.equal(donor.balance(), donorBalance + 300000);
+
+    // should claim if goal is reached'
+    appArgs = [stringToBytes('donate')];
+
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    txGroup = [
+      {
+        type: TransactionType.CallNoOpSSC,
+        sign: SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: {},
+        appArgs: appArgs
+      },
+      {
+        type: TransactionType.TransferAlgo,
+        sign: SignType.SecretKey,
+        fromAccount: donor.account,
+        toAccountAddr: escrow.address,
+        amountMicroAlgos: 7000000,
+        payFlags: {}
+      }
+    ];
+    // execute transaction
+    await runtime.executeTx(txGroup, program, []);
+
+    appArgs = [stringToBytes('claim')];
+    txGroup = [
+      {
+        type: TransactionType.CallNoOpSSC,
+        sign: SignType.SecretKey,
+        fromAccount: creator.account,
+        appId: applicationId,
+        payFlags: {},
+        appArgs: appArgs
+      },
+      {
+        type: TransactionType.TransferAlgo,
+        sign: SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: creator.address,
+        amountMicroAlgos: 0,
+        lsig: escrow,
+        payFlags: { closeRemainderTo: creator.address }
+      }
+    ];
+    await runtime.executeTx(txGroup, program, []);
+    // TODO- close account and tranfer funds to closeRemainderTo in algorand-js
```

Comment:
Please link this TODO with a task on Pivotal (add a link in a pivotal tracker task).

Reply:
ok